cypress-tests/cypress/e2e/PaymentTest/00023-ConnectorAgnostic.cy.js

```diff
@@ -0,0 +1,250 @@
+import * as fixtures from "../../fixtures/imports";
+import State from "../../utils/State";
+import { payment_methods_enabled } from "../PaymentUtils/Commons";
+import getConnectorDetails, * as utils from "../PaymentUtils/Utils";
+let globalState;
+describe("Connector Agnostic Tests", () => {
+  before("seed global state", () => {
+    cy.task("getGlobalState").then((state) => {
+      globalState = new State(state);
+    });
+  });
+
+  after("flush global state", () => {
+    cy.task("setGlobalState", globalState.data);
+  });
+  context(
+    "Connector Agnostic Disabled for Profile 1 and Enabled for Profile 2",
+    () => {
+      let should_continue = true;
+
+      beforeEach(function () {
+        if (!should_continue) {
+          this.skip();
+        }
+      });
+
+      it("Create Business Profile", () => {
+        cy.createBusinessProfileTest(
+          fixtures.createBusinessProfile,
+          globalState
+        );
+      });
+
+      it("connector-create-call-test", () => {
+        cy.createConnectorCallTest(
+          "payment_processor",
+          fixtures.createConnectorBody,
+          payment_methods_enabled,
+          globalState
+        );
+      });
+
+      it("Create Customer", () => {
+        cy.createCustomerCallTest(fixtures.customerCreateBody, globalState);
+      });
+
+      it("Create Payment Intent", () => {
+        let data = getConnectorDetails(globalState.get("connectorId"))[
+          "card_pm"
+        ]["PaymentIntentOffSession"];
+        let req_data = data["Request"];
+        let res_data = data["Response"];
+        cy.createPaymentIntentTest(
+          fixtures.createPaymentBody,
+          req_data,
+          res_data,
+          "no_three_ds",
+          "automatic",
+          globalState
+        );
+        if (should_continue)
+          should_continue = utils.should_continue_further(res_data);
+      });
+
+      it("Confirm Payment", () => {
+        let data = getConnectorDetails(globalState.get("connectorId"))[
+          "card_pm"
+        ]["SaveCardUseNo3DSAutoCaptureOffSession"];
+        let req_data = data["Request"];
+        let res_data = data["Response"];
+        cy.confirmCallTest(
+          fixtures.confirmBody,
+          req_data,
+          res_data,
+          true,
+          globalState
+        );
+        if (should_continue)
+          should_continue = utils.should_continue_further(res_data);
+      });
+
+      it("List Payment Method for Customer using Client Secret", () => {
+        cy.listCustomerPMByClientSecret(globalState);
```

Comment:
should check the count of payment methods

Reply:
once creds issue is resolved we can add

Reply:
please create an issue for this and attach it here as TODO
```suggestion
        // TODO: #1234 add check for count of payment methods
```

Reply:
TODO issues : https://github.com/juspay/hyperswitch-cloud/issues/7339